### ISSUE TITLE
Help me! Some problems about submitting the results of multi-target tracking test set to Kitti's server

### ISSUE THREAD

**Issue by lipeng-gu** [ZwwWayne/mmMOT#5]
Hi，
    I can see from Kitti's website that your multi-target tracking indicators are very good，but I meet some problems about submitting the results of multi-target tracking test set to Kitti's server. I have sent an email to your mailbox. I wonder if you have seen it
    That porblem is that the data in your zip archive submitted to Kitti's server has the false format.
    I have submitted the results of multi-target tracking test sets to the server twice, according to the document requirements, but the server still says that the data in your zip archive has the false format. I can't find the problems. Because I am writing a paper, I am very anxious to solve this problem. I have attached the submitted compressed file. If you are free, can you help me to see if there is any problem with my file format?  Or can you provide a document in the correct format that you submitted？
thank you very much!

**Comment by ZwwWayne** [comment]
Hi @axiaoxiaobai ,
I did not find attachments in your email.
You can check the label of the tracking data, they should have the same format.
The results *.txt should directly inside the zip file.

**Comment by lipeng-gu** [comment]
@ZwwWayne Sorry, I forgot to add the attachment just now. Do you have time to have a look now.

**Comment by ZwwWayne** [comment]
![image](https://user-images.githubusercontent.com/40779233/73430388-e8420a00-4378-11ea-903a-d8b0f716b728.png)
Maybe you should change those -1s to 0.

**Comment by ZwwWayne** [comment]
I am not sure whether you modified this codebase. If you did not, then the function [`write_kitti_result`](https://github.com/ZwwWayne/mmMOT/blob/542f58163d6e90af5ae9d9792c5e6455b318f189/utils/data_util.py#L80) should be called and the results should be OK.

**Comment by lipeng-gu** [comment]
@ZwwWayne thank you. Is the picture on the left the result of your submission？

**Comment by ZwwWayne** [comment]
Yes. And I suggest you also check the tracking devkit and the instructions in the benchmark.

**Comment by lipeng-gu** [comment]
@ZwwWayne 
thank you very much！ I try again.

**Comment by lipeng-gu** [comment]
@ZwwWayne 
Can I have a look at the results of your later columns?

**Comment by lipeng-gu** [comment]
@ZwwWayne 
hi，I submitted the result again, but it was still in the wrong format. Can you send a txt file of a sequence of txt files to my mailbox(2181330@mail.dhu.edu.cn)？

**Comment by ZwwWayne** [comment]
![image](https://user-images.githubusercontent.com/40779233/73511960-a8376180-4422-11ea-9c70-62527eb994c6.png)
You should read and check the readme.txt carefully, and compare the formats between your results and the tracking labels.

**Comment by lipeng-gu** [comment]
@ZwwWayne I read and check the readme.txt carefully，and use writeLabels.m for writing your results. I submitted the results to the server, but the server replied that the file I submitted was in the wrong format.You can send the result of a sequence to my email（2181330@mail.dhu.edu.cn）, and I would like to thank you very much

**Comment by ZwwWayne** [comment]
This is one of the sequence in the submission files.
[0001.txt](https://github.com/ZwwWayne/mmMOT/files/4137576/0001.txt)

**Comment by lipeng-gu** [comment]
@ZwwWayne 
Thank you very much!
I've compared your results with mine. I'd like to ask you a few more questions.
1)Are the unused values are set to 0.0000?
2)All your values retain four decimal, such as 1021.7770. Why does the value in the fifth column only retain one decimal, such as 0.0, or it doesn't matter?
3)Some of the values in the 17th column are set to 0.000, but some are set to -0.0001 and -0.0000. What's the difference?
![image](https://user-images.githubusercontent.com/48339331/73513265-dec3ab00-4427-11ea-8122-1662a5e34934.png)
I'm confused now. Thank you very much!

**Comment by lipeng-gu** [comment]
@ZwwWayne 
There's one problem. If there are several frames without targets, what do you do? Do you want to skip these frames in the txt file without recording any data?

**Comment by ZwwWayne** [comment]
> @ZwwWayne
> There's one problem. If there are several frames without targets, what do you do? Do you want to skip these frames in the txt file without recording any data?

Yes, just skip them.

**Comment by lipeng-gu** [comment]
@ZwwWayne 
Thank you very much! If you are free, can you take a look at the following questions for me?
I've compared your results with mine. I'd like to ask you a few more questions.
1)Are the unused values are set to 0.0000?
2)All your values retain four decimal, such as 1021.7770. Why does the value in the fifth column only retain one decimal, such as 0.0, or it doesn't matter?
3)Some of the values in the 17th column are set to 0.000, but some are set to -0.0001 and -0.0000. What's the difference?

I'm confused now. Thank you very much!

**Comment by ZwwWayne** [comment]
1. Yes.
2. It should do not matter. There has been a while since the code is written, so I cannot remember the reason. But it should works because I use them to submit the results.
3. Same reply as 2.
A good way to debug your results is to use the [`devkit`](https://s3.eu-central-1.amazonaws.com/avg-kitti/devkit_tracking.zip) to evaluate your predictions on the validation set or the total training set and see what happens. If your results could be evaluated by the devkit successfully, you should consider other problems or email the benchmark manager for help.

**Comment by lipeng-gu** [comment]
@ZwwWayne 
Thank you very much.
 I can debug my results by using the devkit to evaluate your predictions on the total training set successfully. But when I submit results to the server, the server always replies that my file format is wrong. So I am confused! I try again.
